src/NzbDrone.Core/Authentication/UserService.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Linq;
+using System.Xml.Linq;
+using Growl.Connector;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.EnvironmentInfo;
+using NzbDrone.Common.Exceptions;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Lifecycle;
+using NzbDrone.Core.Messaging.Events;
+
+namespace NzbDrone.Core.Authentication
+{
+    public interface IUserService
+    {
+        User Add(string username, string password);
+        User Update(User user);
+        User Upsert(string username, string password);
+        User FindUser();
+        User FindUser(string username, string password);
+        User FindUser(Guid identifier);
+    }
+
+    public class UserService : IUserService, IHandle<ApplicationStartedEvent>
+    {
+        private readonly IUserRepository _repo;
+        private readonly IAppFolderInfo _appFolderInfo;
+        private readonly IDiskProvider _diskProvider;
+
+        public UserService(IUserRepository repo, IAppFolderInfo appFolderInfo, IDiskProvider diskProvider)
+        {
+            _repo = repo;
+            _appFolderInfo = appFolderInfo;
+            _diskProvider = diskProvider;
+        }
+
+        public User Add(string username, string password)
+        {
+            return _repo.Insert(new User
+                                {
+                                    Identifier = Guid.NewGuid(),
+                                    Username = username,
```

Comment:
We'd want to use SHA256 probably (so its not reversible). While I agree we should hash it, that would require UI changes that I don't want to time on only to spend more time on it later if we decide to actually implement multiple users. Since right now its just a simple username/password field thats saved with the model.

Its an easy migration to fix the issue later.